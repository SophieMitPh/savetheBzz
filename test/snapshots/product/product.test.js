const { app, Product, request } = require('../../commonJest');

describe('Product Index', function () {
	const productData = {
		name: 'Jeans',
		description: 'Some awesome Jeans',
		price: 55,
	};
	let productProps;
	beforeAll(function (done) {
		Product.create(productData)
			.then(created => {
				productProps = created.Product;
				done();
			})
			.catch(error => {
				console.log('error caught: ' + error.message);
				done(error.message);
			});
	});
	it('renders product detail page correctly', async (done) => {

		console.log('__________________' + productData.name + '__________________________');
		request(app)
			.get('/products/Jeans')
			.then((response) => {
				expect(response.text).toMatchSnapshot();
				done();
			});
	});
});