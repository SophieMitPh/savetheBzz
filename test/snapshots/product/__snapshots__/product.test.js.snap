// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Index renders product detail page correctly 1`] = `
"<link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/css/layout.css\\">
<link href=\\"//fonts.googleapis.com/css2?family=Raleway\\" rel=\\"stylesheet\\">
<link rel=\\"stylesheet\\" href=\\"/css/bootstrap.css\\">
<script type=\\"text/javascript\\" src=\\"/js/jquery.min.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/js/bootstrap.min.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/js/bzzApp.js\\"></script>
<script src=\\"/socket.io/socket.io.js\\"></script>
<body>
    <div class=\\"header-rectangle\\">
    <div class=\\"header-elements\\">
        <div class=\\"headline\\">
            <a href=\\"/\\"><img id=\\"logo\\" src=\\"/images/logo.png\\" /></a>
            <h1 id=\\"headerTitle\\">Save the Bzz!</h1>
        </div>
        <div class=\\"logos\\">
            <div class=\\"login\\">
                
                <a href=\\"/users/login\\">Log In</a>
                <a href=\\"/users/login\\"><img id=\\"account\\" src=\\"/images/account.png\\" /></a>
                
            </div>
            <a href=\\"/wishlist\\"><img id=\\"wishList\\" src=\\"/images/wishList.png\\" /></a>
            <a href=\\"/cart\\"><img id=\\"cart\\" src=\\"/images/cart.png\\" /></a>
            <a href=\\"/chat\\" class=\\"chat-icon\\"><img id=\\"chat-icon-header\\" src=\\"/images/chat.png\\" /></a>
        </div>
    </div>
</div>
    <li><button id=\\"modal-button\\" type=\\"button\\" data-toggle=\\"modal\\"
        data-target=\\"#myModal\\">Added products</button>
<div id=\\"myModal\\" class=\\"modal fade\\" role=\\"dialog\\">
    <div class=\\"modal-dialog\\">
        <h4 class=\\"modal-title\\">Added products:</h4>
        <div class=\\"modal-body\\">
        </div>
        <div class=\\"modal-footer\\">
            <button type=\\"button\\" data-dismiss=\\"modal\\">Close</button>
        </div>
    </div>
</div></li>
    <div class=\\"flashes\\">
        
        
        
    </div>
    
    </div>
   
    <!DOCTYPE html>
<html>

<head>
    <meta charset=\\"utf-8\\">
    <title>Product detailed view</title>
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/css/header.css\\">
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/css/product.css\\">
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"//fonts.googleapis.com/css?family=Rock+Salt\\" />
    <link href=\\"//fonts.googleapis.com/css2?family=Raleway\\" rel=\\"stylesheet\\">
</head>

<body>
    <div class=\\"main\\">
        <div class=\\"productTitle\\">
            <h1>Jeans</h1>
        </div>
        <div class=\\"detailView\\">
        <div class=\\"imgBorder\\">
                <img class=\\"product-image\\" src=\\"/images/Jeans.jpg\\">
        </div>
        <div class=\\"description-container\\">
                <h2>Product info</h2>
                <p>Some awesome Jeans</p>
                <h3>Price: 55 €</h3>
                <button class=\\"add-button\\">ADD TO BAG</button>
        </div>
        </div>
    </div>
</body>

</html>
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/css/footer.css\\">
<footer class=\\"footer\\">
    <div class=\\"footer-container\\">
        <div>
            <p><a class=\\"footer-link\\" href=\\"/users/new\\">Sign Up</a></p>
            <p><a class=\\"footer-link\\" href=\\"/\\">Contact</a></p>
            <p><a class=\\"footer-link\\" href=\\"/\\">Policy Terms</a></p>
        </div>
        <div>
            <p>Authors: MayyaBareeva & Sophie Preußer</p>
            <p><a class=\\"footer-link\\" href=\\"mailto:hege@example.com\\">savethebzz@example.com</a></p>
        </div>
        <div>
            <p><a class=\\"footer-link\\" href=\\"subscribers/new\\">Newsletter</a></p>
        </div>
    </div>
</footer>
</body>"
`;
